pttdb/errors.go

```diff
@@ -25,4 +25,5 @@ var (
 	ErrInvalidLock     = errors.New("invalid db lock")
 	ErrBusy            = errors.New("db busy")
 	ErrInvalidKeys     = errors.New("invalid db keys")
+	ErrInvalidIndex    = errors.New("invalid db index")
```

Comment:
What's the difference between db key and db index? can you add comments on what they are?

Reply:
ErrInvalidIndex is the error when Index cannot be Unmarshalled.
ErrInvalidKeys is the error when the 2nd-indexing-key is invalid in LDBBatch.GetByIdxKey and GetKeyByIdxKey.


In LDBBatch.TryPutAll and LDBBatch.ForcePutAll:

idxKey, idx, and kvs intends to implement 2nd-indexing in leveldb.

Given the id-key and the data with updateTS,
idxKey is the id-key, key is the key-with-updateTS, value is the corresponding data.
Besides, we also have the 2nd-idx-keys to index by the 2nd-idx-keys.
The value of the 2nd-idx-keys are key.

idx includes the key (in the 0th of idx.Keys) and 2nd-idx-keys
kvs are the corresponding key-vals based on the order of idx.